Makefile.defaults

```diff
@@ -5,11 +5,19 @@ ifeq (, $(shell command -v $(YQ_COMMAND) 2> /dev/null))
 $(error "$(YQ_COMMAND) not found, consider doing snap install yq")
 endif
 
-YQ_MAJOR_VERSION := $(shell $(YQ_COMMAND) --version | sed -nE 's/.* v?([0-9]+).*/\1/p')
+# As we need `yq` to be installed to fetch the values from versions.yaml
+# these values are hardcoded here, all other versions/references should be
+# stored in versions.yaml
+YQ_VERSION := v4.35.1
+YQ_CHECKSUM := "sha256:bd695a6513f1196aeda17b174a15e9c351843fb1cef5f9be0af170f2dd744f08"
```

Comment:
Can't we move this into `versions.yaml` ?

Reply:
Do you think we should add a new section for remote files/binaries in the versions.yaml? i.e. At the moment the `tools` section doesn't have a field for checksums. Obviously YAML is flexible that we could just do that the `yq` ...

```yaml
tools:
  golang: 1.20.8
  rust: 1.72.0
  protoc: 3.11.4
  yq:
    version: ...
    checksum: ...
```

Although we don't have a strict schema to follow be nice to keep it consistent.

Reply:
Your right. Let's skip it then. I think the versions.yaml thing won't be for every anyway, as it just can't replace a full build system.